pkg/controller/serviceaccount/legacy_serviceaccount_token_cleaner.go

```diff
@@ -0,0 +1,274 @@
+
+	secretLister         listersv1.SecretLister
+	secretInformerSynced cache.InformerSynced
+
+	podLister         listersv1.PodLister
+	podInformerSynced cache.InformerSynced
+
+	configMapLister         listersv1.ConfigMapLister
+	configMapInformerSynced cache.InformerSynced
+
+	legacySATokenCleanUpInterval time.Duration
+	legacySATokenCleanUpPeriod   time.Duration
+}
+
+// NewLegacySATokenCleaner returns a new *NewLegacySATokenCleaner.
+func NewLegacySATokenCleaner(saInformer coreinformers.ServiceAccountInformer, secretInformer coreinformers.SecretInformer, podInformer coreinformers.PodInformer, configMapInformer coreinformers.ConfigMapInformer, cl clientset.Interface, options LegacySATokenCleanerOptions) (*LegacySATokenCleaner, error) {
+	tc := &LegacySATokenCleaner{
+		client:                       cl,
+		saLister:                     saInformer.Lister(),
+		saInformerSynced:             saInformer.Informer().HasSynced,
+		secretLister:                 secretInformer.Lister(),
+		secretInformerSynced:         secretInformer.Informer().HasSynced,
+		configMapLister:              configMapInformer.Lister(),
+		configMapInformerSynced:      configMapInformer.Informer().HasSynced,
+		podLister:                    podInformer.Lister(),
+		podInformerSynced:            podInformer.Informer().HasSynced,
+		legacySATokenCleanUpPeriod:   options.LegacySATokenCleanUpPeriod,
+		legacySATokenCleanUpInterval: options.LegacySATokenCleanUpInterval,
+	}
+
+	return tc, nil
+}
+
+func (tc *LegacySATokenCleaner) Run(ctx context.Context) {
+	defer utilruntime.HandleCrash()
+
+	logger := klog.FromContext(ctx)
+	logger.Info("Starting legacy service account token cleaner controller")
+	defer logger.Info("Shutting down legacy service account token cleaner controller")
+
+	if !cache.WaitForNamedCacheSync("legacy-service-account-token-cleaner", ctx.Done(), tc.saInformerSynced, tc.secretInformerSynced, tc.configMapInformerSynced, tc.podInformerSynced) {
+		return
+	}
+
+	go wait.UntilWithContext(ctx, tc.evaluateSATokens, tc.legacySATokenCleanUpInterval)
+
+	<-ctx.Done()
+}
+
+func (tc *LegacySATokenCleaner) evaluateSATokens(ctx context.Context) {
+	logger := klog.FromContext(ctx)
+	secretList, err := tc.secretLister.Secrets(metav1.NamespaceAll).List(labels.Everything())
+	if err != nil {
+		logger.Error(err, "Getting secret list from secret cache")
+		return
+	}
+	for _, secret := range secretList {
+		if secret.Type != v1.SecretTypeServiceAccountToken {
+			continue
+		}
+		sa, saErr := tc.getServiceAccount(secret)
+		switch {
+		case saErr != nil:
+			logger.Error(saErr, "Getting service account")
+		case sa == nil:
+			logger.V(4).Info("Service account does not exist", "secret", klog.KRef(secret.Namespace, secret.Name))
+		case tc.shouldCleanToken(logger, secret, sa):
+			logger.Info("Auto-generated service account token is not used for more than 365 * 24 hours, deleting token", "secret", klog.KRef(secret.Namespace, secret.Name))
```

Comment:
don't hard-code the `365 * 24 hours` interval, but do output the creationTimestamp and lastUsed value recorded on the secret